package/kernel/linux/modules/other.mk

```diff
@@ -643,14 +643,16 @@ define KernelPackage/ramoops
   TITLE:=Ramoops (pstore-ram)
   DEFAULT:=m if ALL_KMODS
   KCONFIG:=CONFIG_PSTORE_RAM \
-	CONFIG_PSTORE_CONSOLE=y
+	CONFIG_PSTORE_CONSOLE=y \
+	CONFIG_IO_STRICT_DEVMEM=n
```

Comment:
I think this should be approached this differently.
Best would be to work out if this is expected, otherwise what is not working (talk with upstream maintainer if need be).
Then, if disabling `IO_STRICT_DEVMEM` is required, set it at (sub)target level.
The way the OpenWrt build system works, kmod `=y` or `=n` settings get applied to the kernel config whenever the kmod is selected, and all kmods are selected by the buildbots. As is with this PR, it makes it much less clear that the generic config setting would be overwritten:
https://github.com/openwrt/openwrt/blob/fa6bd065ddad1a3e5d507a4eb9b5fd1fd4b4055a/target/linux/generic/config-6.6#L2739

Reply:
I wasn't aware of the implications of `=y` or `=n` on one kmod in regard to the build system due to all of them being selected. Thanks for taking the time to explain it. That's a bit of a problem then.

I don't think you'll ever want to disable `IO_STRICT_DEVMEM` for a (sub)target because a kmod, that the vast majority doesn't use, needs it.

What I know about the behaviour is that enabling `IO_STRICT_DEVMEM` doesn't seem to restrict writing to the region reserved for `ramoops`. But you can't see anything if that makes sense. How do I know that? You may ask. From a cold booted state I can flash a firmware with `IO_STRICT_DEVMEM` enabled, crash the thing, see that `/sys/fs/pstore` is seemingly empty, flash a firmware with `IO_STRICT_DEVMEM` disabled and find the previous crash log under `/sys/fs/pstore` because the router hasn't been switched off and the memory hasn't been cleared yet.

So there are a couple of things at least that I would love to know. One would be if this is widespread or it only affects some targets. If the former, analysing the differences would be the first step of course.

Another one would be if `ls` and `cat` are not the right tools for the job of retrieving or if they could be invoked in such way that is compatible with `IO_STRICT_DEVMEM` enabled.

I've tried to get some input [here](https://forum.openwrt.org/t/get-ramoops-mtdoops-or-mtdpstore-working-on-more-devices/212316/6) and there is also [#15688](https://github.com/openwrt/openwrt/pull/15688)

Do you think that some of the developers could check if `ramoops` works on their devices?